pack.go

```diff
@@ -178,6 +218,71 @@ func packImage(ctx context.Context, pusher content.Pusher, configMediaType strin
 	return manifestDesc, nil
 }
 
+// packImage packs the given blobs, generates an image manifest for the pack,
+// and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
+// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md#guidelines-for-artifact-usage
+func packImage(ctx context.Context, pusher content.Pusher, artifactType string, layers []ocispec.Descriptor, opts PackOptions) (ocispec.Descriptor, error) {
+	var configDesc ocispec.Descriptor
+	if opts.ConfigDescriptor != nil {
+		configDesc = *opts.ConfigDescriptor
+	} else {
+		// use the empty descriptor for config
+		configDesc = ocispec.DescriptorEmptyJSON
+		configDesc.Annotations = opts.ConfigAnnotations
+		configData := ocispec.DescriptorEmptyJSON.Data
+		if err := pusher.Push(ctx, configDesc, bytes.NewReader(configData)); err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
+			return ocispec.Descriptor{}, fmt.Errorf("failed to push config: %w", err)
+		}
+	}
+	if artifactType == "" {
+		// artifactType MUST be set when config.mediaType is set to the empty value
+		artifactType = configDesc.MediaType
+		if configDesc.MediaType == ocispec.MediaTypeEmptyJSON {
+			artifactType = MediaTypeUnknownArtifact
```

Comment:
Why not returning error? Looks like it's not oras-go's job to define a such a default value for artifact type.

Reply:
You are right! The reason why we used `MediaTypeUnknownArtifact` as the default value for packing Artifact Manifest was because the `artifactType` field was a `SHOULD`. However, it's a MUST for Image Manifest with an empty config.
So we should return an error here.

Reply:
Updated. Thanks for catching this!